symbulate/distributions.py

```diff
@@ -118,19 +131,38 @@ class Hypergeometric(Distribution):
     """
 
     def __init__(self, n, N0, N1):
-        self.n = n
-        self.N0 = N0
-        self.N1 = N1
         
+        if n > 0:
+            self.n = n
+        else:
+            raise Exception("Number of draws cannot be negative")
+        
+        if N0 > 0:
+            self.N0 = N0
+        else:
+            raise Exception("Number of failures cannot be negative")
+        
+        if N1 > 0:
+            self.N1 = N1
+        else:
+            raise Exception("Number of successes cannot be negative")
+
         params = {
             "M" : N0 + N1,
             "n" : N1,
             "N" : n
             }
+
+        if (N0 + N1) < n:
+            raise Exception("Number of Successes + Failures cannot be less than the sample size N")
```

Comment:
Replace "sample size N" by "sample size n". (Uppercase N refers to the population size.)